### PR TITLE
add v1.0.0 kfdef for all platforms with new kusotmize tool

### PR DESCRIPTION
**Pull request by kunmingg**
* publish kfdef for all kubeflow platforms
* get rid of `_` from kfdef names as it's not valid k8s name char
* regenerate kdfef files by https://github.com/kubeflow/manifests/tree/master/kfdef/source
* This PR creates v1.0.0 versions of all the existing KFDef changes
  * The only difference between the master and v1.0.0 versions is that the 1.0.0 versions
     point at the v1.0-branch of kubeflow/manifests

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/826)

### CODE REVIEW

**Comment by richardsliu** [comment]
@kunmingg Do we have to rename the files? Can we just replace the job name when we deploy? For example: https://github.com/kubeflow/kfctl/blob/master/py/kubeflow/kfctl/testing/ci/kfctl_e2e_workflow.py#L346

**Comment by kunmingg** [comment]
@richardsliu 
updated

**Comment by jlewi** [comment]
@kunmingg It looks like the formatting is changed. In particular I see a lot of dictionaries being collapsed to one line
e.g.

```
parameters:
      - name: namespace	      - {name: namespace, value: cert-manager}
```

Not sure why running yaml_dump is giving different output from what I get.

I'd like to minimize the diff from existing config files.

**Comment by kunmingg** [comment]
@jlewi 
updated

with kustomize build one side effect is the comments in kfdef is gone.

**Comment by jlewi** [comment]
LGTM but lets try to get an LGTM from the respective KFDef owners.

@animeshsingh  - Can you or someone else LGTM for IBM
@Jeffwan - Can you provide an LGTM for AWS
@yanniszark  and @krishnadurai -  Can you provide an LGTM for kfctl_k8s_istio.yaml and kfctl_istio_dex.yaml (not sure who is the primary owner for which).

@kunmingg yeah the loss of comments is unfortunate but the alternative maintenance burden of the specs is worse. 

One thing that would be easy to do would be to define a header file for each config that would contain the comments. We could then easily add that as a header to the YAML files when they are generated.

File: kfdef/kfctl_aws.yaml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 apiVersion: kfdef.apps.kubeflow.org/v1
 kind: KfDef
 metadata:
-  name: kubeflow-aws
```
Comment (Jeffwan) [inline kfdef/kfctl_aws.yaml]: name is not needed anymore?

Comment (Jeffwan) [reply kfdef/kfctl_aws.yaml]: Never mind. I see it's name here.  `kfdef/source/v1.0.0/kfctl_aws_cognito.yaml`

File: kfdef/kfctl_istio_dex.yaml [hunk 2]
```diff
@@ -1,10 +1,6 @@
-# This is the config to install Kubeflow on an existing K8s cluster, with support
```
[the remark this answers is not in the document]
Comment (jlewi) [reply kfdef/kfctl_istio_dex.yaml]: @krishnadurai see my comment earlier about preserving comments. If you'd like to preserve comments your immediate option is to manually maintain the kfdef/kfctl_istio_dex.yaml files and update them yourselves. If you want to go that route let @kunmingg know and he will remove the files from this PR.

File: kfdef/kfctl_k8s_istio.yaml [hunk 3]
```diff
@@ -1,14 +1,9 @@
-# This is the config to install Kubeflow on an existing k8s cluster.
```
[the remark this answers is not in the document]
Comment (jlewi) [reply kfdef/kfctl_k8s_istio.yaml]: @krishnadurai see comment above; if the comments need to be pertained you'll either need to maintain the files manually or else update the scripts somehow to preserve comments. See my reply earlier to @kunmingg . We might be able to easily add a header to the file with comments but preserving inline comments when processing YAML is hard.

**Comment by jlewi** [comment]
@krishnadurai the RC for 1.0 is already overdue. So I think there are two options

1. You can LGTM the changes in this PR as is; i.e. with the loss of comments.
1. We can drop kfctl_k8s_istio.yaml from this PR and you can manually update and maintain it in order to preserve comments.

**Comment by krishnadurai** [comment]
@jlewi let us keep the current kustomize generation as is. I'll update the website docs to reflect the information within these comments. I understand the urgency.

I've opened an issue to track the same: kubeflow/website#1592

**Comment by jlewi** [comment]
Thanks @krishnadurai 
/lgtm
/approve

**Pull request merged**